BuildMasterAutomation/Functions/Get-BMDeployment.ps1

```diff
@@ -30,9 +50,35 @@ function Get-BMDeployment
         [Object] $Session,
 
         # The deployment to get. You can pass a deployment id or object.
-        [Parameter(Mandatory)]
         [Alias('ID')]
-        [Object] $Deployment
+        [Object] $Deployment,
+
+        # The application to get deployments for. You can pass an application id, application name, or object.
+        [Object] $Application,
+
+        # The release to get deployments for. You can pass an release id, release name, or object.
+        [Object] $Release,
+
+        # The build to get deployments for. You can pass an build id, build name, or object.
+        [Object] $Build,
+
+        # The environment to get deployments for. You can pass an environment id, environment name, or object.
+        [Object] $Environment,
+
+        # The number for the release to get deployments for.
+        [String] $ReleaseNumber,

```

Comment:
Remove this parameter, I've not even sure where to get the "releaseNumber" from. I've only seen Releases have Id's and Name's.